lib/apiservers/engine/backends/filter/container.go

```diff
@@ -0,0 +1,230 @@
+		id := container.Endpoints[n].ID
+		if listContext.filters.Include("network") {
+			err := listContext.filters.WalkValues("network", func(value string) error {
+				if name == value || id == value {
+					return networkExist
+				}
+				return nil
+			})
+			if err != networkExist {
+				return Exclude
+			}
+		}
+	}
+
+	return Include
+}
+
+/*
+* ValidateContainerFilters will validate the container filters are
+* valid docker filters / values and supported by vic.
+*
+* The function will reuse dockers filter validation
+*
+ */
+func ValidateContainerFilters(options *types.ContainerListOptions, acceptedFilters map[string]bool, unSupportedFilters map[string]bool) (*ContainerListContext, error) {
+	containerFilters := options.Filter
+
+	// ensure filter options are valid and supported by vic
+	if err := ValidateFilters(containerFilters, acceptedFilters, unSupportedFilters); err != nil {
+		return nil, err
+	}
+
+	var filtExited []int
+
+	// we need all containers for these options, so set the All flag
+	if options.Limit > 0 || options.Latest {
+		options.All = true
+	}
+
+	err := containerFilters.WalkValues("exited", func(value string) error {
+		code, err := strconv.Atoi(value)
+		if err != nil {
+			return err
+		}
+		filtExited = append(filtExited, code)
+		return nil
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	err = containerFilters.WalkValues("status", func(value string) error {
+		if !IsValidDockerState(value) {
+			return fmt.Errorf("Unrecognised filter value for status: %s", value)
+		}
+		options.All = true
+		return nil
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	// return value
+	listContext := &ContainerListContext{
+		FilterContext: FilterContext{
+			filters: containerFilters},
+		exitAllowed:          filtExited,
+		ContainerListOptions: options,
+	}
+
+	err = containerFilters.WalkValues("before", func(value string) error {
+		var err error
+		before := cache.ContainerCache().GetContainer(value)
+		if before == nil {
+			err = fmt.Errorf("No such container: %s", value)
+		} else {
+			listContext.BeforeID = &before.ContainerID
+		}
+		return err
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	err = containerFilters.WalkValues("since", func(value string) error {
+		var err error
+		since := cache.ContainerCache().GetContainer(value)
+		if since == nil {
+			err = fmt.Errorf("No such container: %s", value)
+		} else {
+			listContext.SinceID = &since.ContainerID
+		}
+		return err
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return listContext, nil
+}
+
+// DockerState will attempt to transform the passed state
+// to a valid docker state
+// valid states are listed in the func IsValidContainerState
+func DockerState(containerState string) string {
+	var state string
+	switch containerState {
+	case "Stopped":
+		state = "exited"
+	case "Running":
+		state = "running"
+	case "Created":
+		state = "created"
+	default:
+		// not sure what to do, so just return
+		// what was given
+		state = containerState
```

Comment:
I'd rather not -- we know it doesn't match a docker state, so I'd rather leave it as it.